pybossa/view/account.py

```diff
@@ -499,6 +502,15 @@ def create_account(user_data, project_slugs=None, ldap_disabled=True, auto_creat
                      password=user_data['password'])
     msg = generate_invitation_email_for_new_user(user=user_info, project_slugs=project_slugs)
     mail_queue.enqueue(send_mail, msg)
+    data_access = user_data.get('data_access', [])
+    # if the account is created automatically and has L4 data access, send warning
+    if auto_create and 'L4' in data_access:
```

Comment:
`'L4' in data_access` is a necessary but not sufficient condition of invalid firm number.
see [code](https://github.com/bloomberg/pybossa/pull/517/files#diff-ca230f868b3bb408c2217967bf8b5452R118)

Warning emails should not be sent if 'L4' is a result of public instance.

Reply:
I am adding an additional attribute to the user_data to capture the data_access_type in addition to level data.